src/components/Editor/index.tsx

```diff
@@ -39,6 +39,79 @@ import {goToPositionAndCenter} from "./MonacoInstance/utils";
 import "./index.css";
 
 
+/**
+ * Gets the beginning line number of the log event selected by mouse in editor.
+ *
+ * @param editor
+ * @param beginLineNumToLogEventNum
+ * @return the beginning line number of the selected log event.
+ */
+const getSelectedLogEventNum = (
+    editor: monaco.editor.IStandaloneCodeEditor,
+    beginLineNumToLogEventNum: BeginLineNumToLogEventNumMap
+): Nullable<number> => {
+    const selectedLineNum = editor.getPosition()?.lineNumber;
+    if ("undefined" === typeof selectedLineNum) {
+        return null;
+    }
+
+    return getMapValueWithNearestLessThanOrEqualKey(
+        beginLineNumToLogEventNum,
+        selectedLineNum
+    );
+};
+
+/**
+ * Handles copy log event action in the editor.
+ *
+ * @param editor
+ * @param beginLineNumToLogEventNum
+ * @throws Error
+ */
+const handleCopyLogEventAction = (
+    editor: monaco.editor.IStandaloneCodeEditor,
+    beginLineNumToLogEventNum: BeginLineNumToLogEventNumMap
+) => {
+    const selectedLogEventNum = getSelectedLogEventNum(
+        editor,
+        beginLineNumToLogEventNum,
+    );
+
+    if (null === selectedLogEventNum) {
+        return;
+    }
+    const selectedLogEventLineNum =
+        getMapKeyByValue(beginLineNumToLogEventNum, selectedLogEventNum);
+    const nextLogEventLineNum =
+        getMapKeyByValue(beginLineNumToLogEventNum, selectedLogEventNum + 1);
+
+    if (null === selectedLogEventLineNum) {
+        throw new Error("Unable to get the beginning line number of the selected log event.");
+    }
+
+    let endLineNumber: number;
+    if (null !== nextLogEventLineNum) {
+        endLineNumber = nextLogEventLineNum - 1;
+    } else {
+        // Handle the case when this is the last log event in the file.
+        const model = editor.getModel();
+        if (null === model) {
+            throw new Error("Unable to get the text model.");
+        }
+        endLineNumber = model.getLineCount() - 1;
```

Comment:
(just some notes to show we have thought about this - 

i guess we could set `endLineNumber = Infinity` in the case the next the log event is not available. though, it will also select the last empty line, which will be inconsistent with the behaviour of copying other log events

if we really want to make the behaviour consistent (that every log event we copy will carry a newline at the end), we could avoid adjusting the `endLineNumber = nextLogEventLineNum - 1` and set `endColumnNumber = 1`

anyways i don't see our current code / behaviours hurts. i think it's fine to keep everything as-is